app/support/auto_logout.rb

```diff
@@ -13,11 +15,16 @@ def perform
 
   def order_details
     OrderDetail.purchased_active_reservations.
-      where("reservations.actual_end_at IS NULL AND reserve_end_at < ?", Time.zone.now - 1.hour).
+      where("reservations.actual_end_at IS NULL AND reserve_end_at < ?", Time.zone.now).
       includes(:product).
       readonly(false).all
   end
 
+  def auto_logout_time?(reserve_end_at, auto_logout_minutes)
```

Comment:
Combine autologout check into a single method and maybe call it `should_auto_logout?`